apis/risingwave/ng/v1alpha1/risingwave_types.go

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2022 Singularity Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RisingWaveComponentGroupTemplate struct {
+	Image            string                      `json:"image,omitempty"`
+	ImagePullPolicy  corev1.PullPolicy           `json:"imagePullPolicy,omitempty"`
+	ImagePullSecrets []string                    `json:"imagePullSecrets,omitempty"`
+	UpgradeStrategy  appsv1.DeploymentStrategy   `json:"upgradeStrategy,omitempty"`
+	Resources        corev1.ResourceRequirements `json:"resources,omitempty"`
+	NodeSelector     map[string]string           `json:"nodeSelector,omitempty"`
```

Comment:
What labels should node groups have to let selector know?

Reply:
Any labels, I'm not sure what the EKS provides. But I think it's possible to add our own labels.

Reply:
I see, we have to make sure the proper labels are set to those EKS node groups.